core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -1804,7 +1804,8 @@ object SparkContext extends Logging {
    *
    * Access to this field is guarded by SPARK_CONTEXT_CONSTRUCTOR_LOCK
    */
-  private var activeContext: Option[SparkContext] = None
+  private val activeContext: AtomicReference[Option[SparkContext]] = 
```

Comment:
you don't need the Option wrapper anymore for atomic reference. just set it to null when it is empty.